org.eclipse.xtext.util/src/org/eclipse/xtext/util/PolymorphicDispatcher.java

```diff
@@ -210,20 +209,21 @@ public String toString() {
 		}
 	}
 
+	/**
+	 * returns > 0 when o1 is more specific than o2,
+	 * 
+	 * returns == 0 when o1 and o2 are equal or unrelated,
+	 * 
+	 * returns < 0 when o2 is more specific than o1,
+	 */
 	protected int compare(MethodDesc o1, MethodDesc o2) {
-		final List<Class<?>> paramTypes1 = Arrays.asList(o1.getParameterTypes());
-		final List<Class<?>> paramTypes2 = Arrays.asList(o2.getParameterTypes());
-
-		// sort by number of parameters
-		if (paramTypes1.size() > paramTypes2.size())
```

Comment:
Is it safe to remove the check for parameters size? In the following loop you're assuming that the parameters have the same size but if I remember correctly you can specify a polymorphic dispatcher with a minimum and a maximum number of parameters so there might be candidates with different number of parameters, could that be?